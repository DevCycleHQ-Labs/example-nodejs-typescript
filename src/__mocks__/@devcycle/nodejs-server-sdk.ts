const mockClient = {
    variableValue: jest.fn().mockImplementation((user, variableName, defaultValue) => {
        return defaultValue
    })
}

export default {
    initializeDevCycle: jest.fn().mockReturnValue({
        onClientInitialized: jest.fn().mockResolvedValue(mockClient)
    }),
}
