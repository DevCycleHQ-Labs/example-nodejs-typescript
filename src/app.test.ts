import run from "./app";
import request from "supertest";
import { DVCVariableValue } from "@devcycle/nodejs-server-sdk";

let mockDevCycleClient = {
  onClientInitialized: jest.fn(),
  variableValue: jest.fn(),
};
jest.mock("./devcycle", () => ({
  initializeDevCycleClient: () => mockDevCycleClient,
  getDevCycleClient: () => mockDevCycleClient,
}));
jest.mock("./utils/logTogglebot");

describe("greeting", () => {
  const mockVariableValue = (variable: string, value: DVCVariableValue) => {
    mockDevCycleClient.variableValue = jest
      .fn()
      .mockImplementation((user, variableName, defaultValue) =>
        variableName === variable ? value : defaultValue
      );
  };

  test.each(["default", "step-1", "step-2", "step-3"])(
    'returns greeting for variable value "%s"',
    async (value) => {
      mockVariableValue("example-text", value);

      const app = await run();

      return request(app)
        .get("/")
        .then((response) => {
          expect(response.statusCode).toBe(200);
          expect(response.text).toMatchSnapshot();
        });
    }
  );
});
