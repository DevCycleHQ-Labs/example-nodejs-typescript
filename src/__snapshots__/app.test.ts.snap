// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`greeting returns greeting for variable value "default" 1`] = `"<h2>Welcome to DevCycle's example app.</h2><p>If you got to the example app on your own, follow our README guide to create the Feature and Variables you need to control this app in DevCycle.</p><p><a href="/variables">All Variables</a></p>"`;

exports[`greeting returns greeting for variable value "step-1" 1`] = `"<h2>Welcome to DevCycle's example app.</h2><p>If you got here through the onboarding flow, just follow the instructions to change and create new Variations and see how the app reacts to new Variable values.</p><p><a href="/variables">All Variables</a></p>"`;

exports[`greeting returns greeting for variable value "step-2" 1`] = `"<h2>Great! You've taken the first step in exploring DevCycle.</h2><p>You've successfully toggled your very first Variation. You are now serving a different value to your users and you can see how the example app has reacted to this change. Next, go ahead and create a whole new Variation to see what else is possible in this app.</p><p><a href="/variables">All Variables</a></p>"`;

exports[`greeting returns greeting for variable value "step-3" 1`] = `"<h2>You're getting the hang of things.</h2><p>By creating a new Variation with new Variable values and toggling it on for all users, you've already explored the fundamental concepts within DevCycle. There's still so much more to the platform, so go ahead and complete the onboarding flow and play around with the feature that controls this example in your dashboard.</p><p><a href="/variables">All Variables</a></p>"`;
